### PR TITLE
Amp version command - more useful info

### PR DESCRIPTION
**Pull request by JosephGJ**
ref: #548 

Added more useful info to `amp version` command. Based off the `docker version` command. Will display AMP info and Amplifier info when platform is running, only AMP info is displayed if platform is stopped.

More can be added in future.

```
AMP:
 Version:       v0.4.0-dev
 Build:         90dbda1c
 ConfigAddr:    127.0.0.1:8080
 Go version:    go1.7.4
 OS/Arch:       darwin/amd64

Amplifier:
 Version:       v0.4.0-dev
 Build:         unknown
 Port:          :50101
 Go version:    go1.7.4
 OS/Arch:       darwin/amd64
```

### CODE REVIEW

**Review by generalhenry — APPROVED** [review]

File: api/client/amp.go [hunk 1]
```diff
@@ -53,6 +53,8 @@ func (l logger) Println(args ...interface{}) {
 
 // Configuration is for all configurable client settings
 type Configuration struct {
+	Version            string
```
Comment (ndegory) [inline api/client/amp.go]: Should Version and Build be part of the configuration? They will be displayed with the ```amp config``` command, and will be settable (is it a word?) with ```amp config Version value```.

File: api/rpc/version/version.proto [hunk 2]
```diff
@@ -0,0 +1,29 @@
+syntax = "proto3";
+
+package version;
+
+import "google/api/annotations.proto";
+
+service Version {
+  rpc List (ListRequest) returns (ListReply) {
+    option (google.api.http) = {
+      get: "/v1/version",
```
Comment (ndegory) [inline api/rpc/version/version.proto:10]: about the .pb.gw.go, please have a look at the Makefile, the PROTOGWTARGETS are not dynamically set, you should add the new files in this list.

File: api/client/amp.go (hunk 1 above)
Comment (JosephGJ) [reply api/client/amp.go]: I've removed these as they're unnecessary.

File: api/rpc/version/version.proto (hunk 2 above)
Comment (JosephGJ) [reply api/rpc/version/version.proto:10]: I've added `version.pb.gw.go` to the PROTOGWTARGETS.

**Comment by JosephGJ** [comment]
Amplifier's `build` field has been removed (for now) as it is pulling `unknown` from the `Dockerfile` rather than the build from the Makefile.

**Review by subfuzion — APPROVED** [review]
I gave you my feedback in person before the holidays -- nice work, @JosephGJ 👍

**Pull request merged**